pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -436,7 +460,13 @@ func modifyOutboundRouteConfig(push *model.PushContext, in *plugin.InputParams,
 		default:
 			log.Warn("Unknown cluster type in mixer#OnOutboundRouteConfiguration")
 		}
-	case *route.Route_Redirect, *route.Route_DirectResponse:
+	case *route.Route_DirectResponse:
+		if virtualHostname == "block_all" {
```

Comment:
"block_all" should be a constant since it's significant. You can also make a utility function over `route.Route` to detect if it is blackholing. That way you wouldn't have to pass virtualhostname here, and just pass whether the route is blackholing.

Reply:
makes sense, will try your approach.